src/nft/components/bag/BagContent.tsx

```diff
@@ -54,25 +57,47 @@ export const BagContent = () => {
     <>
       <Column display={priceChangedAssets.length > 0 || unavailableAssets.length > 0 ? 'flex' : 'none'}>
         {unavailableAssets.length > 0 && (
-          <UnavailableAssetsHeaderRow
-            assets={unavailableAssets}
-            usdPrice={fetchedPriceData}
-            clearUnavailableAssets={() => setItemsInBag(availableItems)}
-            didOpenUnavailableAssets={didOpenUnavailableAssets}
-            setDidOpenUnavailableAssets={setDidOpenUnavailableAssets}
-            isMobile={isMobile}
-          />
+          <Trace
+            name={EventName.NFT_BUY_BAG_CHANGED}
+            properties={{
+              usd_value: fetchedPriceData,
+              bag_quantity: itemsInBag,
+              ...formatAssetEventProperties(unavailableAssets),
+            }}
+            shouldLogImpression
+          >
+            <UnavailableAssetsHeaderRow
+              assets={unavailableAssets}
+              usdPrice={fetchedPriceData}
+              clearUnavailableAssets={() => setItemsInBag(availableItems)}
+              didOpenUnavailableAssets={didOpenUnavailableAssets}
+              setDidOpenUnavailableAssets={setDidOpenUnavailableAssets}
+              isMobile={isMobile}
+            />
+          </Trace>
+        )}
+        {priceChangedAssets.length > 0 && (
+          <Trace
```

Comment:
would this be better to do with a useeffect?

Reply:
yes, good call